athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/handlers/GlueMetadataHandler.java

```diff
@@ -290,7 +290,7 @@ protected ListTablesResponse doListTables(BlockAllocator blockAllocator, ListTab
         }
         while (nextToken != null);
 
-        return new ListTablesResponse(request.getCatalogName(), tables);
+        return new ListTablesResponse(request.getCatalogName(), tables, null);
```

Comment:
Why didn't you make this paginated? Its part of the SDK and the above code already supports pagination pretty easily. :) What am i missing?

Reply:
The specific connectors' implementation of doListTables calls GlueMetadataHander.doListTables to obtain the list of tables from Glue, then adds additional tables from the actual data source. When a list is an aggregation from more than one data source, pagination based solely on one of the lists, would undoubtedly create duplication of tables. The separate lists must be combined first then paginated. Therefore, pagination should take place in the actual connectors' doListTables implementation after the aggregation yields a full list of tables.

Reply:
Added full pagination support for the GlueMetadataHandler.doListTables API and added tests.